### ISSUE TITLE
Cannot read property 'charAt' of undefined

### ISSUE THREAD

**Issue by TinOo512** [continuationlabs/hapi-capitalize-modules#3]
Hi,

I have an error with ESLint `2.10.2` or with `1.10.3` and the `babel-eslint` parser.

The error `Cannot read property 'charAt' of undefined` appear with this node : 

```
Node {
  type: 'VariableDeclarator',
  start: 63,
  end: 90,
  loc: 
   SourceLocation {
     start: Position { line: 4, column: 6 },
     end: Position { line: 4, column: 33 } },
  range: [ 63, 90 ],
  id: 
   Node {
     type: 'ObjectPattern',
     start: 63,
     end: 71,
     loc: SourceLocation { start: [Object], end: [Object] },
     range: [ 63, 71 ],
     properties: [ [Object] ] },
  init: 
   Node {
     type: 'CallExpression',
     start: 74,
     end: 90,
     loc: SourceLocation { start: [Object], end: [Object] },
     range: [ 74, 90 ],
     callee: 
      Node {
        type: 'Identifier',
        start: 74,
        end: 81,
        loc: [Object],
        range: [Object],
        name: 'require' },
     arguments: [ [Object] ] },
  parent: 
   Node {
     type: 'VariableDeclaration',
     start: 57,
     end: 91,
     loc: SourceLocation { start: [Object], end: [Object] },
     range: [ 57, 91 ],
     declarations: [ [Circular] ],
     kind: 'const',
     parent: 
      Node {
        type: 'Program',
        start: 0,
        end: 711,
        loc: [Object],
        range: [Object],
        body: [Object],
        sourceType: 'script',
        comments: [],
        tokens: [Object],
        parent: null } } }
```

I made a quick fix by changing the line `isCapitalized(node.id.name)` with `isCapitalized(node.id.name || ''))` but I guess there is a better solution.

If you tell me that I should refactor `isRequire` to handle that I can open a PR.

Thanks

**Comment by cjihrig** [comment]
What does the source code that it fails on look like?

**Comment by TinOo512** [comment]
@cjihrig Here is the beginning of the file : 

``` js
const Confidence = require('confidence');
const { pick } = require('ramda');
```

Obviously that fail because of the destructuring... I use the `babel-preset-es2015-node4` to have the few missing ES2015 features.

Do you plan to fix that to be node 6 ready ? Let me know if I should open a PR :)

**Comment by cjihrig** [comment]
Closed in a071cc6d1400ee69662a597313d3b646f1f1fbc4.
